ui/lib/sync/addon/components/secrets/page/destinations/destination/secrets.ts

```diff
@@ -41,15 +41,16 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
       await association.save({ adapterOptions: { action: operation } });
       const action: string = operation === 'set' ? 'Sync' : 'Unsync';
       this.flashMessages.success(`${action} operation initiated.`);
+    } catch (error) {
+      this.flashMessages.danger(`Sync operation error: \n ${errorMessage(error)}`);
+    } finally {
```

Comment:
the route needs to be refreshed whether the update is successful or not so the list view has the correct status badge
![list](https://github.com/hashicorp/vault/assets/68122737/1317e6dc-497c-4c7b-ac5c-b493bf3ad5b5)